pkg/kubelet/server/server.go

```diff
@@ -317,18 +326,28 @@ func (s *Server) InstallAuthFilter() {
 
 		// Get authorization attributes
 		attrs := s.auth.GetRequestAttributes(info.User, req.Request)
+		var allowed bool
+		var msg string
+		for _, attr := range attrs {
+			decision, _, err := s.auth.Authorize(req.Request.Context(), attr)
+			if err != nil {
+				klog.ErrorS(err, "Authorization error", "user", attrs[0].GetUser().GetName(), "verb", attrs[0].GetVerb(), "resource", attrs[0].GetResource(), "subresource", attrs[0].GetSubresource())
+				msg = fmt.Sprintf("Authorization error (user=%s, verb=%s, resource=%s, subresource=%s)", attrs[0].GetUser().GetName(), attrs[0].GetVerb(), attrs[0].GetResource(), attrs[0].GetSubresource())
+				resp.WriteErrorString(http.StatusInternalServerError, msg)
+				return
 
-		// Authorize
-		decision, _, err := s.auth.Authorize(req.Request.Context(), attrs)
-		if err != nil {
-			klog.ErrorS(err, "Authorization error", "user", attrs.GetUser().GetName(), "verb", attrs.GetVerb(), "resource", attrs.GetResource(), "subresource", attrs.GetSubresource())
-			msg := fmt.Sprintf("Authorization error (user=%s, verb=%s, resource=%s, subresource=%s)", attrs.GetUser().GetName(), attrs.GetVerb(), attrs.GetResource(), attrs.GetSubresource())
-			resp.WriteErrorString(http.StatusInternalServerError, msg)
-			return
+			}
+
+			if decision == authorizer.DecisionAllow {
+				allowed = true
+				break
+			}
+
+			klog.V(2).InfoS("Forbidden", "user", attrs[0].GetUser().GetName(), "verb", attrs[0].GetVerb(), "resource", attrs[0].GetResource(), "subresource", attrs[0].GetSubresource())
```

Comment:
I think this should only be logged in the `!allowed` case below.

Reply:
I think the real issue is that I am only ever logging the first attribute. I fixed that now. I do think from a debugging perspective it helps to log the failing request before we try the next one. But if you feel strongly about this I am happy to update it.

Reply:
I feel fairly strongly about only logging this if the request is denied, otherwise people will be wondering why all these forbidden requests are showing up in the kubelet logs, especially without a good way of correlating the requests. I'd be OK with leaving this here at a higher verbosity (5?), and having a lower verbosity one in the overall denied case.

Reply:
I updated the logging as you suggested.